### ISSUE TITLE
Heart Disease Prediction Code Line

### ISSUE THREAD

**Issue by Danmax24** [Danmax24/Python-Class-on-PLP#2]
Complete Code for Heart Disease Prediction
1. Install Required Libraries: Make sure you have the necessary libraries
installed. You can install them using pip:
1. pip install pandas numpy scikit-learn seaborn matplotlib streamlit
2. Create a Python Script: Create a file named heart_disease_prediction.py
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, confusion_matrix
import streamlit as st
# Load the dataset
@st.cache
def load_data():
url =
"https://raw.githubusercontent.com/jbrownlee/Datasets/master/heart-
disease.csv"
data = pd.read_csv(url)
return data
# Preprocess the data
def preprocess_data(data):
# Check for missing values
if data.isnull().sum().any():
data.fillna(data.mean(), inplace=True)
# Split features and target
X = data.drop('target', axis=1)
y = data['target']
# Standardize the features
scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)
return X_scaled, y
# Train the model
def train_model(X, y):
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,
random_state=42)
model = RandomForestClassifier(n_estimators=100, random_state=42)
model.fit(X_train, y_train)
# Evaluate the model
y_pred = model.predict(X_test)
print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))
return model
# Streamlit app
def main():
st.title("Heart Disease Prediction")
# Load data
data = load_data()
st.write("Dataset Overview:")
st.write(data.head())
# Preprocess data
X, y = preprocess_data(data)
# Train model
model = train_model(X, y)
# User input for prediction
st.sidebar.header("User Input")
user_input = []
for column in data.columns[:-1]: # Exclude target column
user_input.append(st.sidebar.number_input(column,
value=float(data[column].mean())))
# Make prediction
if st.sidebar.button("Predict"):
input_data = np.array(user_input).reshape(1, -1)
prediction = model.predict(input_data)
st.write("Prediction: ", "Heart Disease" if prediction[0] == 1 else "No Heart
Disease")
if __name__ == "__main__":
main()
Explanation of the Code
1. Data Loading: The dataset is loaded from a URL. The @st.cache decorator
is used to cache the data for faster loading.
2. Data Preprocessing: The data is checked for missing values, and if any are
found, they are filled with the mean of the respective columns. The
features and target variable are separated, and the features are
standardized.
3. Model Training: A Random Forest Classifier is trained on the data, and the
model's performance is evaluated using a confusion matrix and
classification report.
4. Streamlit App: A simple web app is created using Streamlit, allowing users
to input their medical data and receive a prediction on whether they are at
risk for heart disease.
Explanation of the Code
1. Data Loading: The dataset is loaded from a URL. The @st.cache decorator
is used to cache the data for faster loading.
2. Data Preprocessing: The data is checked for missing values, and if any are
found, they are filled with the mean of the respective columns. The
features and target variable are separated, and the features are
standardized.
3. Model Training: A Random Forest Classifier is trained on the data, and the
model's performance is evaluated using a confusion matrix and
classification report.
4. Streamlit App: A simple web app is created using Streamlit, allowing users
to input their medical data and receive a prediction on whether they are at
risk for heart disease.
Running the Project
1. Save the code in a file named heart_disease_prediction.py.
2. Open a terminal and navigate to the directory where the file is saved.
3. Run the Streamlit app using the following command:
